manta-pay/src/crypto/ecc/arkworks.rs

```diff
@@ -298,7 +300,13 @@ where
         let _ = compiler;
```

Comment:
Yeah, arkworks generates the constraints internally because every variable has a copy of the compiler. ECLAIR uses an explicit compiler for its interfaces so that's why it is an argument here. To use arkworks, we can just drop the borrow of the compiler and let the variables handle it (in fact there is nothing else we can do anyways).